daemon/exec.go

```diff
@@ -248,3 +248,34 @@ func (d *Daemon) Exec(c *Container, execConfig *execConfig, pipes *execdriver.Pi
 
 	return exitStatus, err
 }
+
+// execCommandGC starts a go routine on a ticker to clean up the daemon references
+// of exec configs that are no longer part of the container.
+func (d *Daemon) execCommandGC() {
+	for range time.Tick(5 * time.Minute) {
+		var (
+			cleaned          int
+			liveExecCommands = d.containerExecIds()
+			ids              = d.execCommands.List()
+		)
+		for _, id := range ids {
+			if _, exists := liveExecCommands[id]; !exists {
```

Comment:
I would actually prefer that since the timestamp option is still racey given that we'll never know for sure how long the client will take to query the status of the exec.  With the 'dead list' approach they can take as long as they want, but it forces a model where they don't spin up a ton of execs at once - more of a "one at a time model".  Which I would think you'd prefer because it limits what exec can be used for.